File: Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        stack<ListNode*> stk;
        ListNode * current;
        
        while(head != NULL && head->next != NULL){
            stk.push(head);
            head = head->next;
        }
        
        current = head;
        
        while(!stk.empty()){
            ListNode * tmp = stk.top();
            stk.pop();
            
            tmp->next = NULL;
            current->next = tmp;
            current = current->next;
        }
        
        return head;
    }
};

File: Reverse Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if(!head){
        return head;
    }
    
    const arr = [];
    const answer = new ListNode();
    let point = answer;
    
    while(head){
        arr.push(head.val);
        head = head.next;
    }
    
    for(let i = 0; i < arr.length; i++){
        point.val = arr[arr.length - 1 - i];
        
        if(i < arr.length - 1){
            point.next = new ListNode();
            point = point.next;
        }
    }
    
    return answer;
};